adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.adam.hbase
+
+import java.io.ByteArrayOutputStream
+import org.apache.avro.specific.SpecificDatumWriter
+import org.apache.avro.io.EncoderFactory
+import org.apache.avro.io.DecoderFactory
+import org.apache.avro.io.DatumReader
+import org.apache.avro.io.DatumWriter
+import org.apache.avro.specific.SpecificDatumReader
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable
+import org.apache.hadoop.hbase.io.encoding.DataBlockEncoding
+import org.apache.hadoop.hbase.client.Connection
+import org.apache.hadoop.hbase.client.Admin
+import org.apache.hadoop.hbase.client.ConnectionFactory
+import org.apache.hadoop.hbase.client.Delete
+import org.apache.hadoop.hbase.client.Put
+import org.apache.hadoop.hbase.client.Scan
+import org.apache.hadoop.hbase.client.Get
+import org.apache.hadoop.hbase.client.Table
+import org.apache.hadoop.hbase.client.Result
+import org.apache.hadoop.hbase._
+import org.apache.hadoop.hbase.spark._
+import org.apache.hadoop.hbase.spark.HBaseRDDFunctions._
+import org.apache.hadoop.hbase.util.Bytes
+import org.apache.hadoop.hbase.{ HBaseConfiguration, TableName }
+import org.apache.hadoop.hbase.io.compress.Compression.Algorithm
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.hbase.mapreduce.LoadIncrementalHFiles
+import org.apache.hadoop.fs.{ FSDataInputStream, FileSystem, Path }
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.rdd.variant.{ VariantContextRDD }
+import org.bdgenomics.adam.rich.RichVariant
+import org.bdgenomics.formats.avro._
+import org.bdgenomics.adam.models.{ ReferencePosition, ReferenceRegion, SequenceDictionary, VariantContext }
+import scala.io.Source
+import scala.collection.mutable.ListBuffer
+import scala.collection.JavaConverters._
+import sys.process._
+
+object HBaseFunctions {
+
+  /**
+   * HBase Spark Data Access Object
+   *
+   * In standard usage user will create a single instance of this object, and pass it as the first parameter
+   * when using the HBaseFunctions public API functions
+   *
+   * @param sc SparkContext
+   * @param inConf optional HBase Configuration parameter, only used if manually configuring HBase connection
+   * @param inConnection optional HBase Connection parameter, only used if manually configuring HBase connection
+   * @param inAdmin optional HBase Admin parameter, only used if manually configuring HBase connection
+   * @param inHBaseContext optional HBaseContext parameter, only used if manually configuring HBase connection
+   */
+
+  class HBaseSparkDAO(sc: SparkContext,
```

Comment:
Thoughts on renaming this to something like `HBaseDataAccessObject`? When I first saw `DAO` in the code, I was confused. I don't think it's an obvious abbreviation. I agree that the long name would be a bit verbose, but I tend to prefer verbose names, since characters are cheap but brain time is expensive. ;) 

Also, a small nit: I feel like it's a bit redundant to say `Spark`, but I don't have a strong feeling here.